File: src/main/java/com/spring/cvgenerator/dto/DataObject.java
package com.spring.cvgenerator.dto;

import com.spring.cvgenerator.model.BaseModel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.io.Serializable;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@SuperBuilder
public class DataObject implements Serializable {

    private List<? extends BaseModel> list;

    private Long totalRecords;

    private Integer recordsPerPage;

    private Integer dataSize;

    protected final Integer firstPage = 1;

    private Integer maxPage;

    private Integer currentPage;

}

File: src/main/frontend/cvGenerator/src/app/models/util/data-object.ts
import {BaseModel} from "../base-model";

export interface DataObject<T extends BaseModel> {

  totalRecords: number;
  list: Array<T>;
  recordsPerPage: number;
  dataSize: number;
  firstPage: number;
  maxPage: number;
  currentPage?: number;
}
